FreeRTOS/FreeRTOS
# Updated HTTP S3 download demo to show SigV4 usage

Description
-----------
This PR updated HTTP S3 download demo to show SigV4 usage. Previously the demo uses a pre-signed URL to download from S3. This PR changes the flow to first connecting to IoT credential provider server to obtain a temporary credential, and then connecting to S3 server to download to file with authorization generated using SigV4 library.

Test Steps
-----------
Tested in WinSim. Download worked correctly and logs look ok.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
